modules/rubiconBidAdapter.js

```diff
@@ -1047,12 +1039,30 @@ function applyFPD(bidRequest, mediaType, data) {
       });
     });
     Object.keys(impData).forEach((key) => {
-      (key === 'adserver') ? addBannerData(impData[key].adslot, name, key) : addBannerData(impData[key], 'site', key);
+      if (key !== 'adserver') {
+        addBannerData(impData[key], 'site', key);
+      } else if (impData[key].name === 'gam') {
+        addBannerData(impData[key].adslot, name, key)
+      }
     });
+
+    // add in gpid
+    if (gpid) {
+      data['p_gpid'] = gpid;
+    }
+
+    // only send one of pbadslot or dfp adunit code (prefer pbadslot)
+    if (data['tg_i.pbadslot']) {
+      delete data['tg_i.dfp_ad_unit_code'];
+    }
```

Comment:
this is only relevant to Banner?

Reply:
Yeah, the rubicon PBS adapter will do the handling when it sends to XAPI.

Which is nice so we can just dump stuff into PBS and let the server handle it rather than having a YUGE bid adapter!